tests/common/TestRuntime.cs

```diff
@@ -111,6 +113,24 @@ public static void IgnoreInCI (string message)
 		NUnit.Framework.Assert.Ignore (message);
 	}
 
+	public static void IgnoreInInterpreter (string message)
+	{
+		// until System.Runtime.CompilerServices.RuntimeFeature.IsSupported("IsDynamicCodeCompiled") returns a valid result, atm it
+		// always return true, try to build an object of a class that should fail without introspection, and catch the exception to do the
+		// right thing
+		bool in_interpreter = false;
+		try {
+			AssemblyName aName = new AssemblyName ("DynamicAssemblyExample");
+			AssemblyBuilder ab = AppDomain.CurrentDomain.DefineDynamicAssembly (aName, AssemblyBuilderAccess.RunAndSave);
+			in_interpreter = true;
+		} catch (PlatformNotSupportedException) {
+			// we do not have the interpreter, lets continue
+		}
+		if (!in_interpreter)
+			return;
+		NUnit.Framework.Assert.Ignore (message);
+	}
```

Comment:
that seems to differ from other checks...
we normally have a `bool Check...` to have conditional tests
and a `void Assert...` to ignore tests

it would also be nice if the `bool CheckInterpreter (...)` would be re-used in `tests/EmbeddedResources/ResourcesTest.cs`

Reply:
@spouliot I followed the implementation used in the IgnoreInCI. I have no preference, maybe IgnoreInCI is outdated? If that is the case I can change both if needed.

Reply:
TestRuntime is not available in the EmebededResources project, that is why I did the copy of the logic.

Reply:
Did you try adding the TestRuntime.cs file to the EmbeddedResources project? That would be a better solution that duplicating the code.

Reply:
it's minor but can't we add/link the file to the project ?

I like the `Check`/`Assert` pattern better but that's also minor